frontend/src/components/SettingsDisableNetworkItem/SettingsDisableNetworkItem.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect } from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+import { makeStyles } from '@material-ui/core/styles'
+import { ApplicationState, Dispatch } from '../../store'
+import { SettingsListItem } from '../SettingsListItem'
+import { colors } from '../../styling'
+import { emit } from '../../services/Controller'
+
+export const SettingsDisableNetworkItem: React.FC = () => {
+  const css = useStyles()
+  const { preferences } = useSelector((state: ApplicationState) => ({
+    preferences: state.backend.preferences,
+  }))
+
+  const subLevelTextValue = (
+    <span>
+      This will bind the desktop UI to localhost so that it will only be available through sharing with remote.it
+    </span>
+  )
+  const [disabledLocalNetwork, setDisabledLocalNetwork] = React.useState(preferences.disabledLocalNetwork)
+  const [sublevelText, setSubLevelText] = React.useState(subLevelTextValue)
+
+  const handleClick = () => {
+    emit('preferences', { ...preferences, disabledLocalNetwork: !preferences.disabledLocalNetwork })
+  }
+
+  useEffect(() => {
+    const icon = preferences.disabledLocalNetwork ? 'wifi-slash' : 'wifi'
+
+    if (disabledLocalNetwork == preferences.disabledLocalNetwork) {
+      setSubLevelText(
+        <span>
+          This will bind the desktop UI to localhost so that it will only be available through sharing with remote.it
+        </span>
+      )
+    } else {
+      setSubLevelText(<span className={css.span}>Please restart for changes to take effect.</span>)
+    }
+  }, [preferences.disabledLocalNetwork])
```

Comment:
Hey just an optimization, but you don't need to use an effect here you can set the SublevelText and icon just based on the `disabledLocalNetwork` state in line and not in an effect.